pkg/k8s/wait.go

```diff
@@ -42,7 +42,12 @@ type object interface {
 
 // WaitUntilReady watches for mutations of the target object until the target is ready.
 func WaitUntilReady(ctx context.Context, client rest.Interface, resource string, target object) error {
-	lw := GetListWatch(ctx, client, resource, target)
+	lw := GetListerWatcher(ctx, client, resource, target)
+	if lw == nil {
+		// the ListerWatcher can be nil for tests, simulate waiting by blocking
```

Comment:
Can't this return an implementation that does the blocking instead? (ie don't make the caller aware of the special case)?

Reply:
This is a work in progress. I pushed it up to see if the CI would be happy.

Reply:
done